admin/services/pin.service.js

```diff
@@ -33,4 +36,17 @@ pinService.getActiveSchool = async (schoolId) => {
   return school
 }
 
+/**
+ * Expire pupil's pin
+ * @param token
+ * @param checkCode
+ */
+pinService.expirePupilPin = async (token, checkCode) => {
+  const decoded = jwtService.decode(token)
+  const pupil = await pupilDataService.findOne({_id: ObjectId(decoded.sub)})
+  const currentTimeStamp = moment.utc()
+  await checkDataService.update({checkCode: checkCode}, { '$set': { checkStartedAt: currentTimeStamp } })
+  await pupilDataService.update(pupil._id, { pinExpiresAt: currentTimeStamp })
```

Comment:
Should the pin be reset to null as well?

Reply:
@jon-shipley introduced as part of the update